init-distro-rid.sh

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env bash
+
+# initNonPortableDistroRid
+#
+# Input:
+#   isCrossBuild? (nullable vararg:int)
+#
+# Return:
+#   None
+#
+# Notes:
+#
+# initNonPortableDistroRid will attempt to initialize a non portable rid. These
+# rids are specific to distros need to build the product/package and consume
+# them on the same platform.
+#
+# Below is the list of current non-portable platforms.
+#
+# Builds from the following *must* be non-portable:
+#
+#   |    OS     |           Expected RID            |
+#   -------------------------------------------------
+#   |   rhel6   |           rhel.6-x64              |
+#   |  alpine*  |        linux-musl-(arch)          |
+#   |  freeBSD  |        freebsd.(version)-x64      |
+#
+# It is important to note that the function does not return, but it will set
+# __DistroRid if there is a non-portable distro rid to be used.
+#
+initNonPortableDistroRid()
+{
+    # Make sure out parameter is cleared.
+    export __DistroRid=
+
+    local nonPortableBuildID=""
```

Comment:
I think we should use `__DistroRid` instead of `nonPortableBuildID`, and just export it at the end.

Reply:
This leads to a super weird export __DistroRid=${__DistroRid}. I think I will keep it as it is now. Another local should not be an issue.

Reply:
If `__DistroRid` is already set, you can export it by just doing `export __DistroRid`. Or keep the local - your call.

Reply:
I believe this is resolved with no export until the end